pipeline.py

```diff
@@ -180,318 +185,397 @@ def pipeline(
+            return run_mt_bench_task
+
+        def final_eval_stage():
+            final_eval_task = run_final_eval_op(
+                candidate_model="/output/phase_2/model/hf_format/candidate_model",
+                # TODO: DO we need both candidate_branch and base_branch
+                base_branch=sdg_repo_branch,
+                candidate_branch=sdg_repo_branch,
+                base_model_dir="/model/",
+                max_workers=final_eval_max_workers,
+                merge_system_user_message=final_eval_merge_system_user_message,
+                few_shots=final_eval_few_shots,
+                batch_size=final_eval_batch_size,
+            )
+            mount_pvc(
+                task=final_eval_task,
+                pvc_name=output_pvc_task.output,
+                mount_path="/output",
+            )
+            mount_pvc(
+                task=final_eval_task,
+                pvc_name=sdg_input_pvc_task.output,
+                mount_path="/input",
+            )
+            mount_pvc(
+                task=final_eval_task,
+                pvc_name=model_pvc_task.output,
+                mount_path="/model",
+            )
+
+            use_config_map_as_env(
+                final_eval_task,
+                JUDGE_CONFIG_MAP,
+                dict(endpoint="JUDGE_ENDPOINT", model="JUDGE_NAME"),
+            )
+
+            final_eval_task.set_env_variable("HOME", "/tmp")
+            final_eval_task.set_env_variable("HF_HOME", "/tmp")
+            set_image_pull_secrets(final_eval_task, [IMAGE_PULL_SECRET])
+
+            # uncomment if updating image with same tag
+            # set_image_pull_policy(final_eval_task, "Always")
+
+            use_secret_as_env(
+                final_eval_task, JUDGE_SECRET, {"api_key": "JUDGE_API_KEY"}
+            )
+
+            final_eval_task.after(run_mt_bench_task)
+            final_eval_task.set_accelerator_type("nvidia.com/gpu")
+            final_eval_task.set_accelerator_limit(1)
+
+            return final_eval_task
+
+        def outputs_to_artifacts():
+            output_model_task = pvc_to_model_op(
+                pvc_path="/output/phase_2/model/hf_format/candidate_model",
+            )
+            output_model_task.after(run_mt_bench_task)
+            output_model_task.set_caching_options(False)
+            mount_pvc(
+                task=output_model_task,
+                pvc_name=output_pvc_task.output,
+                mount_path="/output",
+            )
+
+            output_mt_bench_task = pvc_to_mt_bench_op(
+                pvc_path="/output/mt_bench_data.json",
+            )
+            output_mt_bench_task.after(run_mt_bench_task)
+            mount_pvc(
+                task=output_mt_bench_task,
+                pvc_name=output_pvc_task.output,
+                mount_path="/output",
+            )
+
+        # Pipelines
+
+        # some pre-filght image that checks all params and fails if something is wrong.
```

Comment:
Maybe we can revive https://github.com/opendatahub-io/ilab-on-ocp/pull/21 efforts?

Reply:
That's a good idea.